packages/test/test-version-utils/src/versionUtils.ts

```diff
@@ -285,10 +285,42 @@ export function checkInstalled(requested: string) {
  */
 export const loadPackage = async (modulePath: string, pkg: string): Promise<any> => {
 	const pkgPath = path.join(modulePath, "node_modules", pkg);
-	const pkgJson: { main: string } = JSON.parse(
+	// Because we put legacy versions in a specific subfolder of node_modules (.legacy/<version>), we need to reimplement
+	// some of Node's module loading logic here.
+	// It would be ideal to remove the need for this duplication (e.g. by using node:module APIs instead) if possible.
+	const pkgJson: { main?: string; exports?: string | Record<string, any> } = JSON.parse(
 		readFileSync(path.join(pkgPath, "package.json"), { encoding: "utf8" }),
 	);
-	return import(pathToFileURL(path.join(pkgPath, pkgJson.main)).href);
+	// See: https://nodejs.org/docs/latest-v18.x/api/packages.html#package-entry-points
+	let primaryExport: string;
+	if (pkgJson.exports !== undefined) {
+		// See https://nodejs.org/docs/latest-v18.x/api/packages.html#conditional-exports for information on the spec
+		// if this assert fails.
+		// The v18 doc doesn't mention that export paths must start with ".", but the modern docs do:
+		// https://nodejs.org/api/packages.html#exports
+		for (const key of Object.keys(pkgJson.exports)) {
+			if (!key.startsWith(".")) {
+				throw new Error(
+					"Conditional exports not supported by test-version-utils. Legacy module loading logic needs to be updated.",
+				);
+			}
+		}
+		if (typeof pkgJson.exports === "string") {
+			primaryExport = pkgJson.exports;
+		} else {
+			const exp = pkgJson.exports["."];
+			primaryExport = typeof exp === "string" ? exp : exp.require.default;
```

Comment:
This is where I added a fix (this change is coming from https://github.com/microsoft/FluidFramework/pull/18485).
Basically, the `exports["."]` didn't always return just a string and was sometimes an object. This seems to line up if you look at the `package.json` of some dds packages: https://github.com/microsoft/FluidFramework/blob/main/packages/dds/cell/package.json#L22

Reply:
We'll want to do this in main as well, right? And maybe add null chaining for `exp?.require?.default`?

Reply:
Yeah, most likely. I'm not sure what the best fix is here, I just wanted to unblock this integration as I need changes from `main`.